ext/openssl/ossl_config.c

```diff
@@ -136,8 +136,8 @@ config_initialize(int argc, VALUE *argv, VALUE self)
     VALUE filename;
 
     /* 0-arguments call has no use-case, but is kept for compatibility */
-    rb_scan_args(argc, argv, "01", &filename);
     rb_check_frozen(self);
+    rb_scan_args(argc, argv, "01", &filename);
```

Comment:
I guess this diff wasn't intentional?

Reply:
a bit unrelated to the rest, but I tried to align this with what's happening in other similar methods, i.e. avoid the work of parsing args when the object is frozen anyway. Should I revert?

Reply:
This function had no other changes when I added this comment.

I think, if we want to make the order consistent, the frozen state should be checked after the method arity since that's what happens with methods in a C extension that don't take optional parameters or methods written in Ruby. It doesn't matter much in reality and nobody cared about it enough to pick it up, though.

Reply:
reverted